Assets/Packages/com.studio23.ss2.objectivesystem/Runtime/Core/ObjectiveTask.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using NaughtyAttributes;
+using Newtonsoft.Json;
+using Studio23.SS2.InventorySystem.Data;
+using Studio23.SS2.ObjectiveSystem.Data;
+using UnityEngine;
+
+namespace Studio23.SS2.ObjectiveSystem.Core
+{
+    [CreateAssetMenu(menuName = "Studio-23/Objective System/Task", fileName = "objective task")]
+    [Serializable]
+    public class ObjectiveTask : ItemBase
+    {
+        [SerializeField] ObjectiveBase _parentObjective;
+        // we do not actually need to serialize this in editor
+        // this will be different based on save data.
+        // we prevent serialization to avoid merge conflicts
+        // This is manually serialized when saved as needed.
+        [ShowNonSerializedField]
+        private ObjectiveTaskState _state;
+        [SerializeField] int _priority;
+        public int Priority => _priority;
+        public ObjectiveBase ParentObjective => _parentObjective;
+        [ShowNativeProperty]
+        public bool IsActive => _state == ObjectiveTaskState.InProgress || _state == ObjectiveTaskState.Completed;
+        [ShowNativeProperty]
+        public bool IsCompleted => _state == ObjectiveTaskState.Completed;
+        //this is just for the button
+        public bool ObjectiveManagerExists => ObjectiveManager.Instance != null;
+        public bool CompleteParentObjectiveOnCompletion = false;
+
+        public event Action<ObjectiveTask> OnTaskCompletionToggle;
+        public event Action<ObjectiveTask> OnTaskActiveToggle;
+
+        public void SetObjective(ObjectiveBase parentObjective)
+        {
+            this._parentObjective = parentObjective;
```

Comment:
This is so that the objective SO can set the parent objective of hints/tasks in its list. Otherwise, you'd have to remember to set both SO values in inspector which is unnecessary and error prone.